src/gsi/gsimod.F90

```diff
@@ -1978,6 +1989,15 @@ subroutine gsimain_initialize
      endif
   endif
 
+  if (innov_use_model_fed .and. .not.if_model_fed) then
+     if(mype==0) write(6,*)' GSIMOD: invalid innov_use_model_fed=.true. but if_model_fed=.false.'
+     call die(myname_,'invalid innov_use_model_fed,if_model_fed, check namelist settings',332)
+  end if
+
+  if (miter > 0 .and. if_model_fed .and. .not. fed_exist) then
```

Comment:
Should also consider lobserver=.true. ?
namely `if(miter>0 .or. not.lobserver).and.if_model_fed.and.not.fed_exist`. Same for dbz

Reply:
@TingLei-NOAA 
No. In other's setup, lobserver=F for generateing dbz/fed diags for EnKF. That is why I hesitate to add this check in GSI.

Reply:
@hongli-wang  here I meant that lobserver=.true. is still a valid value to start a gsi observer run as far as I know. Hence, 
it should be as " if (.not. (miter == 0 .or. lobserver) .and.(if_model_fed .and. .not. fed_exist)) then " .

Reply:
And, that will be great if you could set up the similar check for dbz .

Reply:
@TingLei-NOAA 
Regarding dbz, I will add @wangym1111 for a best solution. 

@wangym1111 
The current dbz_exist check in wrf_var_mod, doesn't check if the dbz is in anavinfo section: the metguess vector. I added such a check in wrf_var_mod:
if(.not.dbz_exist .or. .not.fed_cloud_exist )then
  dbz_exist=.false.
endif
What do you think? 
As Ting also suggested to add a check for dbz DA. It is like
if (.not. (miter == 0 .or. lobserver) .and.(if_model_dbz .and. .not. dbz_exist)) then
call die to stop GSI run. 

Any thoughts? 
Thanks, 
Hongli

Reply:
> @hongli-wang here I meant that lobserver=.true. is still a valid value to start a gsi observer run as far as I know. Hence, it should be as " if (.not. (miter == 0 .or. lobserver) .and.(if_model_fed .and. .not. fed_exist)) then " .

@TingLei-NOAA 
My current setup keep the option be valid when users want to use the fed from background to calculate  innovation. In other words, when (miter=0, or lobserver=T) and if_model_dbz=T, and dbz_exist=T., FED innovation can be calculated with FED in background, instead of calculating from graupel and other model variables.

Reply:
@hongli-wang I agree with both changes for dBZ. It is much appreciated if you could make the changes for dBZ in this PR along with yours.

Reply:
> > @hongli-wang here I meant that lobserver=.true. is still a valid value to start a gsi observer run as far as I know. Hence, it should be as " if (.not. (miter == 0 .or. lobserver) .and.(if_model_fed .and. .not. fed_exist)) then " .
> 
> @TingLei-NOAA My current setup keep the option be valid when users want to use the fed from background to calculate innovation. In other words, when (miter=0, or lobserver=T) and if_model_dbz=T, and dbz_exist=T., FED innovation can be calculated with FED in background, instead of calculating from graupel and other model variables.

@hongli-wang my question is that " miter >0 while lobserver=.true." still means a GSI observer run.  What will happen for this case?